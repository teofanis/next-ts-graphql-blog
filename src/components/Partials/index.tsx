import { Author } from '@interfaces/app.interfaces'
import Image from 'next/image'
import React from 'react'

interface PostedByProps {
  author: Author
}

export const PostedBy = ({ author }: PostedByProps) => {
  return (
    <div className="flex items-center justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8">
      <Image
        src={author.photo.url}
        alt={author.name}
        width={30}
        height={30}
        className="align-middle rounded-full"
      />
      <p className="inline align-middle text-gray-700 ml-2 text-lg">
        {author.name}
      </p>
    </div>
  )
}

export default { PostedBy }
