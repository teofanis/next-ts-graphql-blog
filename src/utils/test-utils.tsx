import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
