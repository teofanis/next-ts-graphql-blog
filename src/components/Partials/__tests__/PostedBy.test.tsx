import { PostedBy } from '@components/Partials'
import { render } from '@utils/test-utils'
import { author } from '../__fixtures__/author.json'

describe('Posted By Test', () => {
  it('should render correctly', () => {
    const { container } = render(<PostedBy author={author} />)
    expect(container).toMatchSnapshot()
  })
})
