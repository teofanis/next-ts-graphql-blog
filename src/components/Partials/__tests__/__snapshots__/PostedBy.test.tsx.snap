// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posted By Test should render correctly 1`] = `
<div>
  <div
    class="flex items-center justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAiIGhlaWdodD0iMzAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Test Author Name"
        class="align-middle rounded-full"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <p
      class="inline align-middle dark:text-white text-gray-700 ml-2 text-lg"
    >
      Test Author Name
    </p>
  </div>
</div>
`;
